compiler/rustc_span/src/lev_distance.rs

```diff
@@ -46,6 +46,49 @@ pub fn lev_distance(a: &str, b: &str, limit: usize) -> Option<usize> {
     (dcol[m] <= limit).then_some(dcol[m])
 }
 
+/// Provides a method name similarity score between two method names. The lower the score,
+/// the closer the match. 0 is an identical match.
+///
+/// Uses the Levenshtein distance between the two strings and attempts to account
+/// for cases where one is a substring of the other.
+///
+/// Returns None if the distance exceeds the limit
+pub fn method_name_similarity_score(a: &str, b: &str, limit: usize) -> Option<usize> {
+    let n = a.chars().count();
+    let m = b.chars().count();
+
+    // Check one isn't less than half the length of the other
+    let big_len_diff = (n * 2) < m || (m * 2) < n;
+    let len_diff = if n < m { m - n } else { n - m };
+    let lev = lev_distance(a, b, limit + len_diff)?;
+
+    let score = lev - len_diff;
+
+    let score = {
+        if score == 0 && len_diff > 0 && !big_len_diff {
+            1
+        } else if !big_len_diff {
+            score + (len_diff + 1) / 2
+        } else {
+            score + len_diff
+        }
+    };
+
+    (score <= limit).then_some(score)
+}
+
+/// Finds the best match for given method name in the given iterator.
+///
+/// A version of [`find_best_match_for_name`] that uses a modified Levenshtein score
+/// to provide more useful matches for method names.
+pub fn find_best_match_for_method_name(
```

Comment:
As before, I think these would be better if they didn't mention method names. It's not obvious what modifications to a Levenshtein score are necessary for method names when this is the only comment you've seen, so it's probably better to describe how it is modified.

Reply:
I think I've removed all the references to method now